generate.js

```diff
@@ -1,59 +1,68 @@
 #!/usr/bin/env node
 
+const { config, pages } = require('../../criticalcss.config');
 const critical = require('critical');
-const { getConfigFile, getClosestHeight } = require('./helpers');
-const criticalcssConfig = getConfigFile();
+const { getClosestHeight } = require('./helpers');
 
-// Disables warnings
+if (!config) {
+    console.warn('WARNING: Project configuration missing, falling back to default configuration');
+}
+
+// Disables warnings.
 process.setMaxListeners(0);
 
-// No configs so stop
-if (!criticalcssConfig) {
+// No pages configured, so stop.
+if (!pages.pages) {
+    throw new Error('Error: You need to provide pages as configuration (e.g. pages: [{"src": "https://www.site.com/", "target": "app/design/frontend/*company*/theme/web/css/cms-critical.css"}])');
     return;
 }
 
-const domain = process.env.DOMAIN || criticalcssConfig?.domain;
-const waitBeforeRender = process.env.RENDER_WAIT_TIME || criticalcssConfig?.renderWaitTime || 2500;
-const penthouseTimeout = process.env.PENTHOUSE_TIMEOUT || criticalcssConfig?.penthouseTimeout || 30000;
-const forceIncludeList = criticalcssConfig?.forceInclude || [];
-const forceExcludeList = criticalcssConfig?.forceExclude || [];
-
-// The pages you want the critical css from (multiple pages and themes possible)
-const pages = criticalcssConfig?.pages || [];
-let dimensions = criticalcssConfig?.dimensions || [];
+const domain = process.env.DOMAIN || pages?.domain;
+if (!domain) {
+    throw new Error('Error: You need to provide a domain to target');
+    return;
+}
 
 // The dimensions you want critical css from.
+let dimensions = config?.dimensions || [];
 dimensions = dimensions?.map((dimension) => ({
     width: dimension?.width,
     height: dimension?.height ?? getClosestHeight(dimension?.width)?.height,
 }));
 
-console.log('Starting to create critical css bundles...');
+const baseConfig = {
```

Comment:
Wat je zou kunnen doen, en ik zelf altijd ook makkelijk vind, is een aparte `config.js` of `config.default.js` file maken waar je de standaard config in zet. Dan kunnen andere developers die ook altijd gebruiken als naslag van wat de mogelijk opties zijn zonder dat ze door de code hoeven te spitten.

`config.default.js`:
```js
module.exports = {
        inline: false,
        penthouse: {
             timeout: 30000,
             renderWaitTime: 2500
        }
       // eventueel als comments alle andere opties
    }
```

En dan bovenaan deze file:
```js
const baseConfig = require('config.default.js');
```

Reply:
Ja goeie, hoewel "alle mogelijke opties" niet zo heel handig is. Die opties zijn namelijk de opties van dependency "criticalcss". Maar wellicht kan ik een verwijzing daar naar maken.